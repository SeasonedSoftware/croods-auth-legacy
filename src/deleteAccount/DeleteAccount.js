import React from 'react'
import { Destroy } from 'croods'

import render from './render'

export default props => {
  const { relativePath = '/auth' } = props
  return (
    <Destroy
      id="currentUser"
      path={relativePath}
      name="auth.deleteAccount"
      parseDestroyResponse={() => ({ destroyed: { id: 'currentUser' } })}
      render={render(props)}
    />
  )
}
