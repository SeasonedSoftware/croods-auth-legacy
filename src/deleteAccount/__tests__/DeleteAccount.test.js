import React from 'react'
import renderer from 'react-test-renderer'
import DeleteAccount from '../DeleteAccount'

jest.mock('croods', () => ({
  Destroy: props => <div {...props}>Destroy - {props.render(props)}</div>,
}))

jest.mock('../render', () => props => destroy => (
  <div {...props} destroy={destroy}>
    renderDestroy
  </div>
))

const props = { foo: 'bar' }
const tree = renderer.create(<DeleteAccount {...props} />)

it('renders correctly', () => {
  expect(tree.toJSON()).toMatchSnapshot()
})

it('parses response correctly', () => {
  const data = { important: true }
  const response = {
    foo: 'bar',
    data,
  }

  const parser = tree.root.findByProps({ name: 'auth.deleteAccount' }).props
    .parseDestroyResponse
  expect(parser(response)).toMatchObject({ destroyed: { id: 'currentUser' } })
})
