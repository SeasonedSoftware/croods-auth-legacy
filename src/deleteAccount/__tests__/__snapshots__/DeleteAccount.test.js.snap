// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="currentUser"
  name="auth.deleteAccount"
  parseDestroyResponse={[Function]}
  path="/auth"
  render={[Function]}
>
  Destroy - 
  <div
    destroy={
      Object {
        "id": "currentUser",
        "name": "auth.deleteAccount",
        "parseDestroyResponse": [Function],
        "path": "/auth",
        "render": [Function],
      }
    }
    foo="bar"
  >
    renderDestroy
  </div>
</div>
`;
